src/test/java/org/broadinstitute/hellbender/tools/examples/ExampleCollectMultiMetricsIntegrationTest.java

```diff
@@ -0,0 +1,68 @@
+package org.broadinstitute.hellbender.tools.examples;
+
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.metrics.MetricAccumulationLevel;
+import org.broadinstitute.hellbender.tools.examples.metrics.multi.ExampleCollectMultiMetricsSpark;
+import org.broadinstitute.hellbender.utils.test.ArgumentsBuilder;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.broadinstitute.hellbender.utils.test.IntegrationTestSpec;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Test for example multi-level metrics collector.
+ */
+public final class ExampleCollectMultiMetricsIntegrationTest extends CommandLineProgramTest {
+    private static final File TEST_DATA_DIR = new File(getTestDataDir(), "/examples/metrics");
+
+    public String getTestedClassName() {
+        return ExampleCollectMultiMetricsSpark.class.getSimpleName();
+    }
+
+    @DataProvider(name="metricsInputFiles")
+    public Object[][] insertSizeMetricsInputFiles() {
+        return new Object[][] {
+                {"exampleMetrics.bam", true, "expectedMultiMetricsL3.txt"},
+                {"exampleMetrics.bam", false, "expectedMultiMetricsL1.txt"},
+        };
+    }
+    @Test(dataProvider="metricsInputFiles", groups = "spark")
+    public void test(
+            final String fileName,
+            final boolean allLevels, // collect metrics for each possible level
+            final String expectedResultsFile) throws IOException {
+
+        final File input = new File(TEST_DATA_DIR, fileName);
+        final File textOut = BaseTest.createTempFile("test", ".txt");
+
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+
+        // IO arguments
+        args.add("-" + StandardArgumentDefinitions.INPUT_SHORT_NAME);
+        args.add(input.getAbsolutePath());
+
+        args.add("-" + StandardArgumentDefinitions.OUTPUT_SHORT_NAME);
+        args.add(textOut.getAbsolutePath());
+
+        if (allLevels) {
+            args.add("-" + "LEVEL");
```

Comment:
You can use 'args.add("MetricsAccumulationLevel", MetricAccumulationLevel.SAMPLE.toString()) which might be less verbose.